pkg/collector/corechecks/network/ntp.go

```diff
@@ -75,8 +77,22 @@ func (c *ntpConfig) parse(data []byte, initData []byte) error {
 	}
 
 	c.instance = instance
-	if c.instance.Host == "" {
-		c.instance.Host = fmt.Sprintf("%d.datadog.pool.ntp.org", rand.Intn(3))
+	if c.instance.Host != "" {
```

Comment:
I don't think you need a distinct case for `c.instance.Hosts == nil`, you can append to the nil slice (ie. https://play.golang.org/p/fMOoZSjlbbY)

Reply:
You're right, removing this code branch